airmozilla/popcorn/static/popcorn/js/editor.js

```diff
@@ -1,2 +1,29 @@
-// Initialize the editor with the div id and path to Popcorn Editor.
-PopcornEditor.init('editor', '/static/popcorn/PopcornEditor/editor.html');
+$(document).ready(function () {
+  function getSlug() {
+    var config = $('slug');
+    if (config.length) {
+      return config.data('url');
+    } else {
+      throw 'Slug not provided in DOM. Cannot get slug';
+    }
+  }
+
+  PopcornEditor.listen(PopcornEditor.events.loaded, function () {
+    $.get('/popcorn/data/', {
+        slug: getSlug()
+    })
+    .done(function (response) {
+      if(response.data) {
+        PopcornEditor.loadInfo(response.data);
+      } else {
+        PopcornEditor.loadInfo(PopcornEditor.createTemplate(response.metadata));
+      }
+    })
+    .fail(function() {
+      console.warn("Unable to load popcorn data :(");
+      console.error.apply(console, arguments);
```

Comment:
These actually don't help the user that much. Only if they know how to enable the web developer web console will the get a hint that if finished loading but didn't work. 

Couldn't you do something like 
```javascript
.fail(function() {
    ...same console.error lines as before...
    $('#load-error').show();
})
```
And in `editor.html` something like
```html
<div id="load-error" style="display:none">
<p>Someone something something</p>
</div>
```